tonic/src/transport/server/mod.rs

```diff
@@ -408,9 +411,9 @@ impl<L> Server<L> {
     /// [eco]: https://github.com/tower-rs
     /// [`ServiceBuilder`]: tower::ServiceBuilder
     /// [interceptors]: crate::service::Interceptor
-    pub fn layer<NewLayer>(self, new_layer: NewLayer) -> Server<NewLayer> {
+    pub fn layer<NewLayer>(self, new_layer: NewLayer) -> Server<Stack<L, NewLayer>> {
         Server {
-            layer: new_layer,
+            layer: Stack::new(self.layer, new_layer),
```

Comment:
Actually now that I think about it. We should use consider using `tower::ServiceBuilder`. Then we ensure the layers are added in the right order. For example I believe the new layer should be added as the left layer in the stack as that is what [`ServiceBuilder::layer`](https://docs.rs/tower/latest/tower/struct.ServiceBuilder.html#method.layer) does.

Reply:
This is a `tower` issue, but unless I'm misunderstanding the code, those docs are wrong.  It seems like it wraps the given layer with the inner layer rather than wrapping the inner layer with the given one.

Anyway, I'll look into using `ServiceBuilder` later tonight or tomorrow.